### DISCUSSION TITLE
Open Media Vault not seeing mounted drive

### DISCUSSION THREAD

**Discussion by ecotechie** [tteck/Proxmox#2554] [Help Needed] [answered]
I ran the OMV script with default options and added the following line to the LXC's conf file:
`mp0: /dev/nvme3n1,mp=/mnt/btr-vault,backup=0`

This allowed the LXC to mount and see the files in that partition:
![image](https://github.com/tteck/Proxmox/assets/11607383/6d5b36fd-ee4f-41f8-b4a4-d71ae28566c4)

However OMV does not see the partition that has been mounted:
![image](https://github.com/tteck/Proxmox/assets/11607383/dc5d34cb-0b78-4a17-8572-8d9c22c08d49)

How would I get OMV to see the mounted partition?

I have read from the OMV side that this is not supported, but it seems it is doable. Otherwise this script wouldn't have been created?

https://docs.openmediavault.org/en/stable/troubleshooting.html
![image](https://github.com/tteck/Proxmox/assets/11607383/7d4d112c-5e4c-4503-bd83-ad7fd64499e5)

https://forum.openmediavault.org/index.php?thread/21269-solutions-to-common-problems/
![image](https://github.com/tteck/Proxmox/assets/11607383/6298c4fc-e8a1-48b0-acda-28bf9383cfed)

https://docs.openmediavault.org/en/stable/installation/index.html
![image](https://github.com/tteck/Proxmox/assets/11607383/e73d8479-c52a-4fd7-8484-82de16b1ce2b)

**Comment by MDBInd** [comment]
As you rightly pointed out OMV does not support LXC and the script should be refactored to create a VM instead.

**Comment by tteck** [comment]
Removed from website.

**Comment by rdlu** [comment, marked as answer]
For me it works wonderfully, but it's definitely not straightforward. I installed around an year ago, but did a lot of manual stuff like this:

```
 cat /var/lib/lxc/150/config

...
lxc.mount.entry = /dev/disk/by-id/ata-ST4000VN006- dev/sda none bind,optional,create=file
lxc.mount.entry = /dev/disk/by-id/ata-ST4000VN006--part1 dev/sda1 none bind,optional,create=file
lxc.mount.entry = /dev/disk/by-id/ata-ST4000DM005- dev/sdb none bind,optional,create=file
lxc.mount.entry = /dev/disk/by-id/ata-ST4000DM005--part1 dev/sdb1 none bind,optional,create=file
lxc.mount.entry = /dev/disk/by-id/ata-ST4000DM004- dev/sdc none bind,optional,create=file
lxc.mount.entry = /dev/disk/by-id/ata-ST4000DM004--part1 dev/sdc1 none bind,optional,create=file
lxc.mount.entry = /dev/disk/by-id/ata-WDC_WD40EFAX-68JH4N1_WD- dev/sdd none bind,optional,create=file
lxc.mount.entry = /dev/disk/by-id/ata-WDC_WD40EFAX-68JH4N1_WD--part1 dev/sdd1 none bind,optional,create=file
...
```

```
cat /var/lib/lxc/150/mount-hook.sh
mknod -m 777 ${LXC_ROOTFS_MOUNT}/dev/sdb b 8 16
mknod -m 777 ${LXC_ROOTFS_MOUNT}/dev/sdb1 b 8 17
mknod -m 777 ${LXC_ROOTFS_MOUNT}/dev/sdc b 8 32
mknod -m 777 ${LXC_ROOTFS_MOUNT}/dev/sdc1 b 8 33
mknod -m 777 ${LXC_ROOTFS_MOUNT}/dev/sda b 8 0
mknod -m 777 ${LXC_ROOTFS_MOUNT}/dev/sda1 b 8 1

# actually this is disabled, you only need the previous, but this was the recommended on several forums posts I found
```

basically mounting the drives directly 

![image](https://github.com/tteck/Proxmox/assets/109541/f55b5797-0539-474e-901b-f0fc3ec90e0f)

Sad to see it go, but I understand it's not a novice thing. Understanding lxc and searching a lot of confusing forum posts and random blogs takes time.

**Reply by tteck** [reply]
Added back to the website.

**Comment by MDBInd** [comment]
Glad you got it working, I had problems trying to use a ZFS volume created in Proxmox in OMV LXC.  I created a new VM and got it up and running on LXC in minutes. As you say I did find an obscure Chinese [blog post ](https://kawsing-gitbook-io.translate.goog/opensystem/andoid-shou-ji/pomoxve/proxmox-ve-+-openmediavault-shi-yong-lxc?_x_tr_sl=zh-TW&_x_tr_tl=en&_x_tr_hl=en&_x_tr_pto=sc&_x_tr_hist=true) that alluded to a solution but its straying far from the path and might be rendered useless in an update so I am happy with the VM solution.

**Comment by oliv99** [comment]
tteck script need to be updated to debian 12 for omv7
I have installed omv7 over LXC debian 12 without any issue as per the official forum
https://forum.openmediavault.org/index.php?thread/50222-install-omv7-on-debian-12-bookworm/

tested with 4 disks without partitions but with md raid5 existing

manual lxc config
lxc.mount.entry: /dev/disk/by-id/ata-HGST_HTS721010A9E630_JR1000--BA6E dev/sdq none bind,optional,create=file
lxc.mount.entry: /dev/disk/by-id/ata-HGST_HTS721010A9E630_JR1000--68XE dev/sdr none bind,optional,create=file
lxc.mount.entry: /dev/disk/by-id/ata-HGST_HTS721010A9E630_JR1000--GHEE dev/sds none bind,optional,create=file
lxc.mount.entry: /dev/disk/by-id/ata-HGST_HTS721010A9E630_JR1004--Y0VM dev/sdt none bind,optional,create=file

fdisk  -l /dev/sd[q-t] [work correctly]
mdadm --assemble --verbose /dev/md9 /dev/sdq /dev/sdr /dev/sds /dev/sdt [do not work]

may be a partition need to be created first

**Comment by oliv99** [comment]
**Second attempt to mount full hard disk inside omv lxc container and to create a raid5 array**

root@pmx6:~#  cat /etc/pve/lxc/1022.conf

lxc.mount.entry: /dev/disk/by-id/ata-HGST_HTS721010A9E630_JR1000--GHEE dev/sde none bind,optional,create=file
lxc.mount.entry: /dev/disk/by-id/ata-HGST_HTS721010A9E630_JR1000--BA6E dev/sdf none bind,optional,create=file
lxc.mount.entry: /dev/disk/by-id/ata-HGST_HTS721010A9E630_JR1000--968XE dev/sdg none bind,optional,create=file
lxc.mount.entry: /dev/disk/by-id/ata-HGST_HTS721010A9E630_JR1004--YM0M dev/sdh none bind,optional,create=file
lxc.mount.entry: /dev/md1 dev/md1 none bind,optional,create=file

I have imported a mdadm raid5 array within omv7
root@omv7:~# mdadm --assemble --verbose /dev/md1 /dev/sde /dev/sdf /dev/sdg /dev/sdh

this is working by adding the last mount entry "/dev/md1"
the dev/sdx identification within LXC must match the host the /dev/sdx of the host

![20240416_snap_1713257491](https://github.com/tteck/Proxmox/assets/134931891/c0ca079d-0d48-4896-9ef3-8e8e249a165a)


even disk smart info are correctly reported

**Reply by rdlu** [reply]
For me the /dev/sdX points are completely different. But I'm not using md arrays anyways, I'm using snapraid.

But yeah, even smart is reported. I was able to replace a drive with a smart alert in due time, that was the same time I started using `/dev/disk/by-id` as a source, because the /dev/sdx got "messed up" with the disk change. 
sdb on omv for instance sometimes is completely different on the proxmox host, depending if I connect an extra disk to the sata ports or not.

The disk in danger is still connected, by I'm using as a secondary cold storage now, with backups that are verifiable, either with rar recovery segments or hash files.

![image](https://github.com/tteck/Proxmox/assets/109541/aff87bde-22bb-4a02-a068-5bdd1f5faeef)

There's one thing that I never made work, that is the backup of the root omv partition with their own plugins, but proxmox is taking care of that.
